lib/db/transactions.go

```diff
@@ -473,11 +475,8 @@ func (t readWriteTransaction) putFile(fkey []byte, fi protocol.FileInfo, truncat
 func (t readWriteTransaction) updateGlobal(gk, keyBuf, folder, device []byte, file protocol.FileInfo, meta *metadataTracker) ([]byte, bool, error) {
 	l.Debugf("update global; folder=%q device=%v file=%q version=%v invalid=%v", folder, protocol.DeviceIDFromBytes(device), file.Name, file.Version, file.IsInvalid())
 
-	var fl VersionList
-	svl, err := t.Get(gk)
-	if err == nil {
-		_ = fl.Unmarshal(svl) // Ignore error, continue with empty fl
-	} else if !backend.IsNotFound(err) {
+	fl, err := t.getGlobalVersionsByKey(gk)
```

Comment:
This changes behaviour, we no longer ignore the error.

Reply:
Yeah, right. However that's actually the "new correct behaviour" since https://github.com/syncthing/syncthing/pull/6411 where I apparently missed this.